iroha_p2p/src/message.rs

```diff
@@ -0,0 +1,46 @@
+use serde::{Deserialize, Serialize};
+use parity_scale_codec::{Decode, Encode};
+use iroha_crypto::Hash;
+use std::io;
+
+#[derive(Debug, Clone, Encode, Decode, iroha_actor::Message)]
+pub enum Message {
+    Error,
+    Data { data: Vec<u8> },
+    ClientHello { key: [u8; 32] },
+    ServerHello { key: [u8; 32], nonce: [u8; 12] }
+}
```

Comment:
It would be better if message will be abstracted over some type which implements encode and decode

Reply:
Also key and nonce should be some generic structure

Reply:
Over-engineering! Don't you think?

Reply:
No, I don't think so. It will be type safe, generic and data will have methods associated with them. Not some random bytes